src/stdlib/String.ts

```diff
@@ -233,11 +234,29 @@ export const Val = new Callable(
     "Val",
     {
         signature: {
-            args: [{name: "s", type: ValueKind.String}],
-            returns: ValueKind.Float
+            args: [
+                {
+                    name: "s", 
+                    type: ValueKind.String
+                },
+                {
+                    name: "radix",
+                    type: ValueKind.Int32,
+                    defaultValue: new Expr.Literal(new Int32(10))
+                }
+            ],
+            returns: ValueKind.Dynamic
         },
-        impl: (interpreter: Interpreter, s: BrsString): Float => {
-            return new Float(Number(s.value));
+        impl: (interpreter: Interpreter, s: BrsString, brsRadix: Int32): BrsNumber => {
+            function isBrsStrFloat(str: BrsString): boolean {
```

Comment:
> NOTE: I have a really cheesy way to detect if a string is a float just be checking for a decimal place. Open to other ideas.

You could probably use [Number.isInteger](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) here in combination with [Number.isNan](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN)?